docs/kb/semgrep-code/scan-engine-kill.md

```diff
@@ -135,10 +129,14 @@ to see both your RAM and your swap space usage in Linux.
 
 ## Parallelization
 
-Once you have determined the RAM sufficient to scan your large codebase, you can now introduce parallelization to speed up the subproccesses that can be parallelized. Larger Kubernetes pod and runners charge on a time basis, so it now becomes a study of feasiblity to achieve the optimal configuration.  
+Once you have determined the RAM sufficient to scan your large codebase, you can now introduce parallelization to speed up the scan.
+
+Per the previous section, you have determined the total memory required for a `-j 1` configuration. Now, you can begin testing different parallelization configurations, such as specifying `-j 2`, all the whilst still monitoring for any swap usage.
+
+If that succeeds, you can experiment with a `-j 4` configuration and continue monitoring scan time for improvements.  Keeping resource costs in mind, it serves well to determine the optimal configuration in bumping up the number of parallel jobs to speed up the scan. Thus, there is a balancing act between achieving faster scans and keeping your overhead costs down as much as possible. 
 
-You can start with a large Kubernetes pod but still only run with one job with `-j 1` set so that you can determine what the total memory is required for this kind of job. Then, you can pare this memory usage somewhat, such as by specifying `-j 2`, all the whilst still monitoring for any swap usage.
+Furthermore, there is overhead in parallelization: the total RAM required for a `-j 2` scan is greater than a `-j 1` scan for the same codebase, but you should see a decrease in total scan time.
 
-If that succeeds, you can experiment with a `-j 4` configuration and continue decreasing the scan time in nearly half and so on.  Since Kubernetes pods are charging per-minute, it serves you well to determine the optimal configuration in bumping up the number of parallel jobs to speed up the scan.  Thus, there is a balancing act between achieving full scan coverage and keeping your overhead costs down as much as possible.  
+## Scanning schedule
 
-You can expect to see more total RAM than `-j 1` but not exactly half per thread. Codebases are highly variant; it's difficult to benchmark `N number of lines per MB`. But there is overhead in parallelization, so you would not get a true halving of RAM required going from 1 to 2 jobs. In fact, the total RAM required is likely to be greater for a `-j 2` job than a `-j 1` job, but you should see a decrease in total time.  This is why the empirical testing is important, to find where the right balance is in total time spent versus cost.
+We recommend performing modular scans on a daily basis along with one comprehensive monolithic scan every two weeks or so to address code introduced outside the scope of the modular scans. This helps true-up your dependency graph for the monolithic build and optimizes scan feasibility.
```

Comment:
I'm not sure if I understand this. What exactly are we recommending, and why?

Reply:
The shortest definition I can give based on our [scan-monorepo-in-parts](https://semgrep.dev/docs/kb/semgrep-ci/scan-monorepo-in-parts/) article: A module implies a top-level or high-level folder whose code files don't interact with the code files in another module. If this is succinct without oversimplifying, maybe @sinat101 can use this definition somewhere early in the doc or reference [https://semgrep.dev/docs/kb/semgrep-ci/scan-monorepo-in-parts/](https://semgrep.dev/docs/kb/semgrep-ci/scan-monorepo-in-parts/)? 

We could also present the recommendation the following way:

---

A modular scan means to scan a top-level folder whose code files do not interact with code files in other folders at the same level. We recommend the following:

- Perform a modular full scan daily by using the following command: `semgrep ci --include=/src/moduleA/*`.
- Perform a full scan on the entire monorepo every two weeks. This full scan ensures that any dependency scans are up-to-date.

---

Regarding the last sentence, I'm not actually sure if a full scan affects dependency scans, tbh, I based that off of the existing content. 

Just my 2 cents :)